crates/sui-graphql-rpc/tests/e2e_tests.rs

```diff
@@ -506,6 +511,168 @@ mod tests {
+                    digest
+                    sender {
+                        address
+                    }
+                    gasInput {
+                        gasSponsor {
+                            address
+                        }
+                        gasPrice
+                    }
+                }
+                error
+            }
+        }"#;
+        let variables = vec![GraphqlQueryVariable {
+            name: "tx".to_string(),
+            ty: "String!".to_string(),
+            value: json!(tx_kind_bytes),
+        }];
+        let res = cluster
+            .graphql_client
+            .execute_to_graphql(query.to_string(), true, variables, vec![])
+            .await
+            .unwrap();
+        let binding = res.response_body().data.clone().into_json().unwrap();
+        let res = binding.get("dryRunTransactionBlock").unwrap();
+
+        let digest = res.get("transaction").unwrap().get("digest").unwrap();
+        // Dry run txn does not have digest
+        assert!(digest.is_null());
+        assert!(res.get("error").unwrap().is_null());
+        let sender_read = res.get("transaction").unwrap().get("sender").unwrap();
+        // Since no transaction metadata is provided, we use 0x0 as the sender while dry running the trasanction
+        // in which case the sender is null.
+        assert!(sender_read.is_null());
+        assert!(res.get("results").unwrap().is_array());
+    }
+
+    // Test that we can handle dry run with failures at execution stage too.
+    #[tokio::test]
+    #[serial]
+    async fn test_dry_run_failed_execution() {
+        let _guard = telemetry_subscribers::TelemetryConfig::new()
+            .with_env()
+            .init();
+
+        let connection_config = ConnectionConfig::ci_integration_test_cfg();
+
+        let cluster =
+            sui_graphql_rpc::test_infra::cluster::start_cluster(connection_config, None).await;
+
+        let addresses = cluster.validator_fullnode_handle.wallet.get_addresses();
+
+        let sender = addresses[0];
+        let coin = *cluster
+            .validator_fullnode_handle
+            .wallet
+            .get_gas_objects_owned_by_address(sender, None)
+            .await
+            .unwrap()
+            .get(1)
+            .unwrap();
+        let tx = cluster
+            .validator_fullnode_handle
+            .test_transaction_builder()
+            .await
+            // A split coin that goes nowhere -> execution failure
+            .move_call(
+                SUI_FRAMEWORK_PACKAGE_ID,
+                "coin",
+                "split",
+                vec![
+                    CallArg::Object(ObjectArg::ImmOrOwnedObject(coin)),
+                    CallArg::Pure(bcs::to_bytes(&1000u64).unwrap()),
+                ],
+            )
+            .with_type_args(vec![GAS::type_tag()])
+            .build();
+        let tx_bytes = Base64::from_bytes(&bcs::to_bytes(&tx).unwrap());
+        let tx_bytes = tx_bytes.encoded();
```

Comment:
I vaguely recall there's a `Base64::encode` that does these two steps in one, although maybe I'm misremembering?